### DISCUSSION TITLE
Ergo Pool - Unable to query/insert into table pgres

### DISCUSSION THREAD

**Discussion by dbaldree** [coinfoundry/miningcore#1019] [Q&A] [answered]
Getting so close to a working pool but now am getting this error when the api gets called. 
I can see the tables in postgres - I can see the owner and grants are good however they are all empty.
Any idea what I missed in the setup? or is something preventing the pool from seeding the tables with the pool configuration.

197340-[2021-12-01 17:52:53.8972] [D] [StatsRepository] InsertPoolStatsAsync()
197413:[2021-12-01 17:52:53.9041] [E] [StatsRecorder] Npgsql.PostgresException (0x80004005): 42703: column "poolid" does not exist

The poolstats table schema seems intact just empty:
 id | poolid | connectedminers | poolhashrate | sharespersecond | networkhashrate | networkdifficulty | lastnetworkblocktime | blockheight | connectedpeers | created
----+--------+-----------------+--------------+-----------------+-----------------+-------------------+----------------------+-------------+----------------+---------
(0 rows)

clearly the column does exist so the error is partly erroneous in itself.

[api.log]
A quick curl test: curl http://127.0.0.1:4000/api/pools === api calls bails with error 500 certainly because there are no rows to retrieve:
 Exception data:
    Severity: ERROR
    SqlState: 42883
    MessageText: operator does not exist: @ text
    Hint: No operator matches the given name and argument type. You might need to add an explicit type cast.
    Position: 285
    File: parse_oper.c
    Line: 722
    Routine: op_error    at Npgsql.Internal.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|213_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
...
[2021-12-01 17:51:26.0299] [I] [Diagnostics] Request finished HTTP/1.1 GET http://127.0.0.1:4000/api/pools - - - 500 0 - 342.4783ms

**Comment by dbaldree** [comment]
Also note I was running latest Postgres and got the issue there first - I downgraded to Postgres 11 based on some threads I found - rebuilt everything in postgres -- but still the same issue.

**Comment by oliverw** [comment]
MC runs fine - and actually prefers - Postgres 14

**Reply by dbaldree** [reply]
OK I will put it back to v14 -- any idea why it won't insert data?
197340-[2021-12-01 17:52:53.8972] [D] [StatsRepository] InsertPoolStatsAsync()
197413:[2021-12-01 17:52:53.9041] [E] [StatsRecorder] Npgsql.PostgresException (0x80004005): 42703: column "poolid" does not exist

**Comment by oliverw** [comment]
That will be fixed in the next release.

**Reply by dbaldree** [reply]
awesome news will give it a whirl... thank you indeed

**Comment by oliverw** [comment, marked as answer]
https://github.com/coinfoundry/miningcore/releases/tag/v64

**Reply by dbaldree** [reply]
perfect! this is resolved thanks very much.
